### PR TITLE
419 initialize terms hack

### PR DESCRIPTION
**Pull request by fenekku**
Let's use `82_metadata_upload_setup` like we use `our_master` in our fork of invenio-deposit: just as a way to integrate our work without stepping on the origin's master.

This is a bit of a hack. We won't be using invenio-records-js in the foreseeable future, so it's fine. It's a hack because it hardcodes some assumptions about the structure of the initial data and because it packages `dist/` with it. The latter is needed because invenio-deposit expects a distributable npm package (easier for me to just include it). Normally we don't want to do these things.

### CODE REVIEW

File: src/invenio-records-js/directives/invenioRecordsForms.js [hunk 1]
```diff
@@ -171,7 +171,23 @@ function invenioRecordsForm($q, schemaFormDecorators, InvenioRecordsAPI,
           // Pre-process the query value/identifier
           query = options.scope.insideModel;
           query = scope.$eval(options.processQuery || 'query', {query: query});
+        } else if (Array.isArray(options.initial) && options.initial.length > 0) {
+          // If there is no query, no previous suggestions, no insideModel
```
Comment (sharpattack) [inline src/invenio-records-js/directives/invenioRecordsForms.js:175]: What is `insideModel`?

File: dist/invenio-records-js.js [hunk 2]
```diff
@@ -0,0 +1,703 @@
...
+          var titleMap = options.initial.map(function(element) {
+            return {
+              "name": "(" + element["source"] + ") " + element['value'],
+              "value": element
+            }
+          });
+          var defer = $q.defer();
+          defer.resolve({"data": titleMap});
+          return defer.promise;
+        }
```
Comment (sharpattack) [inline dist/invenio-records-js.js:232]: The lines from 222-232 are new here, right?

**Review by sharpattack — COMMENTED** [review]
Where are we using this branch, `82_metadata_upload_setup` again? I could not find reference to it or to the most recent commit on that branch. 

Everything in `dist/` is unchanged, except where I specifically ask, right? I get why you did this. 

When adding new terms, or removing the initial values/terms during edit, does the record update properly?

File: src/invenio-records-js/directives/invenioRecordsForms.js (hunk 1 above)
Comment (fenekku) [reply src/invenio-records-js/directives/invenioRecordsForms.js:175]: It's one of the locations where the uiselect library saves data. I don't know why they couldn't use `internalModel`... This is why I was complaining a little last time: this library is using `internalModel`, `externalModel`, `insideModel`, `titleMap` and `model['<json key>']` as ways to store array of "terms" in various formats at various times. They had a reason, I don't know what it was.

We can safely ignore it.

File: dist/invenio-records-js.js (hunk 2 above)
Comment (fenekku) [reply dist/invenio-records-js.js:232]: Yes. The content of `dist/` is all autogenerated from the content of `src/`.

**Comment by fenekku** [comment]
> Where are we using this branch, 82_metadata_upload_setup again? I could not find reference to it or to the most recent commit on that branch.

A long time ago, I experimented with a fork of invenio-records-js (82_metadata_upload_setup) because I couldn't make the latest version of invenio-records-js on npm work. I didn't end up using my tweaks, since it proved to be simpler to just use an earlier version of the npm package that *did* work. Now though, we are using this commit through `invenio-deposit` and it is referred there by its commit hash. So above when I said "integrate our work", I just meant it as a way to get a review - I won't be changing the commit hash in invenio-deposit to the one of the merged commit (thinking out loud: if fast forward rebasing keeps commit hashes, then I wouldn't even have too...)

**Comment by fenekku** [comment]
> When adding new terms, or removing the initial values/terms during edit, does the record update properly?

As far as I could tell yes. I am hoping the next testing session will have more people able to test the cases I didn't test.

**Comment by sharpattack** [comment]
> we are using this commit through `invenio-deposit` and it is referred there by its commit hash

Yes, but because branch `82_metadata_upload_setup` includes a commit we weren't using before, just checking that it's not introducing breaking changes.

**Review by sharpattack — APPROVED** [review]
Looks good, hopefully records updates terms properly too.

**Pull request merged**